### PR TITLE
[dashboard] Adjust Prebuild and Project Configurator pages to spec

### LINKED ISSUE #5392: [configuration] Don't suggest/allow starting another prebuild when one is running
In configuration I can start a prebuild. When that is running tze "Run Prebuild" button is still green and active.
When I click it I loose the current logs and see an error message, telling me that I cannot start a prebuild because one is running already.

### LINKED ISSUE #5501: [configurator] Buttons don't do anything
When looking at the configurator of a git-configured project, the "Run Prebuild" button is disabled.
Also, the "Open Workspace" button doesn't do anything.

### PR DESCRIPTION
**Pull request by jankeromnes**
Aligns the Prebuild & Project Configurator pages with the specs to fix https://github.com/gitpod-io/gitpod/issues/4423

TODO:

- [x] Show detailed Prebuild status and runtime below the logs in the Prebuild and Configurator pages
- [x] Refactor `server.triggerPrebuild` to make `branchName` optional (picks the default branch and makes e.g. the Prebuilds page load faster) and return information about the triggered Prebuild (so that we can get the logs after triggering in the Configurator)
- [x] Refactor the Configurator to use `server.triggerPrebuild` instead of the "manual" prebuild prefix (thus keeping the prebuild/project association)
- [x] Enable `Run Prebuild` button for Git-based configurations in the Configurator
- [x] Fix the `Run Prebuild` and `New Workspace` buttons for the Prebuild & Configurator pages, fixes #5392 and fixes #5501

Drive-by fixes:

- [x] Rename `Detecting project type ...` loading screen to `Detecting project configuration ...` to be more truthful
- [x] Add more discoverable 'change' link when adding projects from GitLab/GitHub

/uncc

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **jankeromnes** after the PR has been reviewed.

Associated issue: *#4423*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**
- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5474#issuecomment-911873911" title="Approved">jankeromnes</a>*

Associated issue: *#4423*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [jankeromnes]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [jankeromnes]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [jankeromnes]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by jankeromnes** [comment]
Okay, sorry that this became so big and took so long to finalize. It is now ready for review.

@svenefftinge @AlexTugarev @gtsiolis Could you please take a look? (Please use GitLab projects, GitHub isn't integrated.)

/cc @svenefftinge
/cc @AlexTugarev 
/cc @gtsiolis

**Comment by jankeromnes** [comment]
Okay, already found two problems:
- [ ] If a Configuration is already saved in the Project (but not in Git), it shouldn't be automatically re-guessed, just restored from the DB
- [ ] When starting a workspace from the Configurator, it no longer seems to have the additional config files locally 😳

**Comment by gtsiolis** [comment]
Looking at this now! 👀

**Comment by jankeromnes** [comment]
> Looking at this now! 👀

Many thanks! 🙏 (Please beware, the PR is currently being re-deployed -- this may cause a brief outage, but shouldn't reset your data or anything you've tested.)

**Comment by gtsiolis** [comment]
> Please beware, the PR is currently being re-deployed -- this may cause a brief outage, but shouldn't reset your data or anything you've tested.

That's my favorite way of reviewing a PR. 🙈

File: components/dashboard/src/projects/ConfigureProject.tsx
Comment (gtsiolis) [inline components/dashboard/src/projects/ConfigureProject.tsx:167]: **question**: This is taking noticeably more time sometimes, like 4-5 seconds or more when there's no git based configuration. Expected?

File: components/dashboard/src/projects/Prebuilds.tsx
Comment (gtsiolis) [inline components/dashboard/src/projects/Prebuilds.tsx:272]: **issue**: This is currently showing duration as seconds like `5s` but this could look confusing when showing time duration longer than a minute. For example, `306s`. Ideally this could surface time duration for `306s` as like `05:06` as described in the initial specs in https://github.com/gitpod-io/gitpod/issues/4948#issuecomment-895927769. However, it should be fine to resolve this in a follow up issue. 🏓 

| 5s | 306s |
|-|-|
| duration-1 | duration-2 |

**Comment by AlexTugarev** [comment]
Code looks good! 💯 

One question to the `New Workspace` button on the Prebuild page: is it practical? You cannot start a workspace for old prebuilds anyways, so the page is more of a view. Wdyt of removing it from there? 

Screen Shot 2021-09-03 at 11 09 34

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5474#issuecomment-912388528" title="Approved">AlexTugarev</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5474#issuecomment-911873911" title="Approved">jankeromnes</a>*

Associated issue: *#4423*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [AlexTugarev,jankeromnes]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [AlexTugarev,jankeromnes]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [AlexTugarev,jankeromnes]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/dashboard/src/projects/Prebuilds.tsx
Comment (gtsiolis) [inline components/dashboard/src/projects/Prebuilds.tsx:288]: **praise**: Nice! This reminded me of [slots in Vue](https://vuejs.org/v2/guide/components-slots.html).

Comment (gtsiolis) [inline components/dashboard/src/projects/Prebuilds.tsx:288]: **question**: Now that we have proper status updates down here and this component is used as is in the project configurator, wondering if it makes sense to change or remove the status from the header although it could be useful to have the status up there.

If we remove the header instance we'll have to make sure the bottom part of the prebuild output is always visible. Of course, we could leave it as is for now but make sure the designs for both is identical. Alternatively, we could only include the status icon to avoid having duplicated information. What do you think? 

| BEFORE | AFTER (A) | AFTER (B) |
|-|-|-|
| BEFORE | AFTER (A) | AFTER (B) |

File: components/dashboard/src/projects/ConfigureProject.tsx
Comment (gtsiolis) [inline components/dashboard/src/projects/ConfigureProject.tsx:99]: **issue**: This is probably out of the scope of these changes but noticed that it takes a while to redirect to the configuration page once you select a team. Is this expected or needed? Cc @AlexTugarev

File: components/dashboard/src/projects/Prebuild.tsx
Comment (gtsiolis) [inline components/dashboard/src/projects/Prebuild.tsx:93]: **issue**: Could not get any prebuild log output here. Expected? 😭

File: components/dashboard/src/projects/Prebuilds.tsx
Comment (gtsiolis) [inline components/dashboard/src/projects/Prebuilds.tsx:266]: **issue**: Noticed that the status on the header section does not automatically get updated but you need to reload to the get new status. This could lead to having a discrepancy between these two places where status is surfaced.

| Example A |  Example B | 
|-|-|
| Frame 218 | Frame 2182 |

**Review by gtsiolis — APPROVED** [review]
One step closed to shipping **Teams & Projects**! 3️⃣ 2️⃣ 1️⃣ 🚀 

Thanks for adding this @jankeromnes! 🏀

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 79a8555afa90a11325b2bb73d460bdaebf992e7a</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5474#issuecomment-912388528" title="Approved">AlexTugarev</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5474#pullrequestreview-745864504" title="Approved">gtsiolis</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5474#issuecomment-911873911" title="Approved">jankeromnes</a>*

Associated issue: *#4423*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [AlexTugarev,jankeromnes]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [AlexTugarev,jankeromnes]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [AlexTugarev,jankeromnes]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by svenefftinge** [comment]
> One question to the `New Workspace` button on the Prebuild page: is it practical? You cannot start a workspace for old prebuilds anyways, so the page is more of a view. Wdyt of removing it from there?

I'd opt to leave it here. I was just using it and didn't have any doubts that it was handy there. It should maybe show what it does (Open workspace on branch `branch-name`) in the tooltip.

File: components/dashboard/src/projects/Prebuild.tsx
Comment (svenefftinge) [reply components/dashboard/src/projects/Prebuild.tsx:93]: It took some time the first time I opened it. But I, at last, got logs. I think this is unrelated to this PR.

File: components/dashboard/src/projects/ConfigureProject.tsx
Comment (svenefftinge) [reply components/dashboard/src/projects/ConfigureProject.tsx:167]: Yes, I think we should look if we can do that ahead of time.

**Review by svenefftinge — APPROVED** [review]
Such an improvement! 💯  I can feel how all this could be useful 😁 

The small nits should be done in small separate PRs. Based on all issues that we assign ourselves before taking them.

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5474#issuecomment-912388528" title="Approved">AlexTugarev</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5474#pullrequestreview-745864504" title="Approved">gtsiolis</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5474#issuecomment-911873911" title="Approved">jankeromnes</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5474#pullrequestreview-746003348" title="Approved">svenefftinge</a>*

Associated issue: *#4423*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [AlexTugarev,jankeromnes,svenefftinge]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [AlexTugarev,jankeromnes,svenefftinge]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [AlexTugarev,jankeromnes,svenefftinge]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/dashboard/src/projects/ConfigureProject.tsx
Comment (jankeromnes) [reply components/dashboard/src/projects/ConfigureProject.tsx:99]: True, @AlexTugarev noticed a UI freeze, and I did too today. Maybe it's indeed MonacoEditor? (It's loaded lazily in the Configurator, in order not to make the entire dashboard slower)

File: components/dashboard/src/projects/Prebuild.tsx
Comment (jankeromnes) [reply components/dashboard/src/projects/Prebuild.tsx:93]: Yes, sometimes logs don't show up at all ("Error: could not find headless logs for 1234"). Sometimes, they don't show up live, but eventually the entire logs show up after the prebuild is complete.

This morning I also had live logs. It's kind of a shame that this isn't reliable.

File: components/dashboard/src/projects/Prebuilds.tsx
Comment (jankeromnes) [reply components/dashboard/src/projects/Prebuilds.tsx:266]: Agreed: The top status is based on the "prebuild" status, while the bottom one is based on the "prebuild workspace instance" status. Normally they should be in sync, but discrepancies can happen during incidents or because of bugs.

**Comment by jankeromnes** [comment]
Many thanks for the awesome feedback! ❤️

I agree with merging this now, and filing/doing the improvements as follow-ups.

> One question to the `New Workspace` button on the Prebuild page: is it practical? You cannot start a workspace for old prebuilds anyways, so the page is more of a view. Wdyt of removing it from there? 

Good point. Like @svenefftinge, I'd prefer keeping it for now, but independently of whether we want to keep it, I believe the link is actually incorrect 😬 (it points to the context URL of the prebuild, which becomes incorrect when you push new commits on top of the branch -- instead, it should probably link to the commit URL instead).

/unhold

File: components/dashboard/src/projects/Prebuild.tsx
Comment (svenefftinge) [reply components/dashboard/src/projects/Prebuild.tsx:93]: We need to look into it (in a separate PR). Can we merge this? 😁

**Comment by svenefftinge** [comment]
> instead, it should probably link to the commit URL instead

It should link to the branch but this needs to be communicated somehow.

**Comment by jankeromnes** [comment]
> But I see the point now. Having a button on an old outdated prebuild would be confusing.

Or, we can still link to the exact (potentially outdated) prebuild by using the commit context URL.

But maybe after using it for a bit we'll know more about what's better here.

File: components/dashboard/src/projects/NewProject.tsx
Comment (jankeromnes) [reply components/dashboard/src/projects/NewProject.tsx:388]: Filed as https://github.com/gitpod-io/gitpod/issues/5524

File: components/dashboard/src/projects/ConfigureProject.tsx
Comment (jankeromnes) [reply components/dashboard/src/projects/ConfigureProject.tsx:167]: We could also probably optimize the "config inferrer": It currently makes repository requests one by one (e.g. "does a `yarn.lock` file exist? Oh... does a `cargo.toml` file exist? Oh... etc.)

Fetching the entire repo structure (not necessarily file content or history) in one go, and using that to respond to the "config inferrer"'s queries would already be a huge performance boost.

File: components/dashboard/src/projects/Prebuilds.tsx
Comment (jankeromnes) [reply components/dashboard/src/projects/Prebuilds.tsx:272]: Filed as https://github.com/gitpod-io/gitpod/issues/5525

File: components/dashboard/src/projects/Prebuilds.tsx
Comment (jankeromnes) [reply components/dashboard/src/projects/Prebuilds.tsx:288]: Excellent point and proposals! Filed as https://github.com/gitpod-io/gitpod/issues/5526

File: components/dashboard/src/projects/ConfigureProject.tsx
Comment (jankeromnes) [reply components/dashboard/src/projects/ConfigureProject.tsx:167]: Now tracked as https://github.com/gitpod-io/gitpod/issues/5527

**Pull request merged**
